tests/plugin.test.ts

```diff
@@ -144,6 +145,49 @@ describe("Purchases", () => {
     );
   });
 
+  describe("setLogLevel", () => {
+    // If the enum is backed by string values, iterating over the enum won’t work because the `value` in our loop is a
+    // string instead of a number. The indexer of `LOG_LEVEL` can only accept `VERBOSE`, `DEBUG`...
+    // We need a helper method to create a new type `enumKeys`
+    for (const value of enumKeys(LOG_LEVEL)) {
+      it(`setLogLevel(${value}) fires PurchasesPlugin with the correct arguments`, () => {
+        Purchases.setLogLevel(LOG_LEVEL[value]);
+        expect(execFn).toHaveBeenCalledWith(
+            null,
+            null,
+            "PurchasesPlugin",
+            "setLogLevel",
+            [value]
+        );
+      });
+    }
+  });
+
+  describe("setLogHandler", () => {
+    // If the enum is backed by string values, iterating over the enum won’t work because the `value` in our loop is a
+    // string instead of a number. The indexer of `LOG_LEVEL` can only accept `VERBOSE`, `DEBUG`...
+    // We need a helper method to create a new type `enumKeys`
+    for (const value of enumKeys(LOG_LEVEL)) {
+      it(`setLogLevel(${value}) fires PurchasesPlugin with the correct arguments`, () => {
+        let receivedLogLevel;
+        Purchases.setLogHandler((logLevel, message) => {
+          receivedLogLevel = logLevel
+          expect(message).toEqual("message");
+        });
+        expect(execFn).toHaveBeenCalledWith(
+            expect.any(Function),
+            null,
+            "PurchasesPlugin",
+            "setLogHandler",
+            []
+        );
+        let capturedCallback = execFn.mock.calls[0][0] as LogHandler;
+        capturedCallback(LOG_LEVEL[value], "a message");
+        expect(receivedLogLevel).toEqual(LOG_LEVEL[value]);
```

Comment:
I am struggling with this test. The callback is executed in plugin.ts, but the arguments are undefined, so when getting to line 174, logLevel and message are also undefined.

Reply:
Hmm so I'm just guessing but when we first set the log handler, we send a result with a "No result" value. Maybe it's picking that up?

Reply:
That part is in the native side which is not being executed since we are mocking `window.cordova.exec`. I am capturing the first parameter of that call (the `capturedCallback`) and invoking it. That part is actually working, becuase if you set a breakpoint in the callback in plugin.ts#952 it stops there, but for some reason the parameters of the callback are `undefined`.